src/components/NumberTicker.tsx

```diff
@@ -0,0 +1,114 @@
+import * as React from 'react'
+import { Animated, StyleProp, StyleSheet, Text, TextStyle, View } from 'react-native'
+import { typeScale } from 'src/styles/fonts'
+
+interface CommonProps {
+  textHeight: number
+  textStyles?: StyleProp<TextStyle>
+  animationDuration?: number
+}
+interface Props extends CommonProps {
+  finalValue: string
+  testID?: string
+}
+
+interface TickProps extends CommonProps {
+  startValue: number
+  endValue: number
+}
+
+interface TickTextProps extends CommonProps {
+  value: string
+}
+
+const numberRange = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9']
+
+function TickText({ value, textHeight, textStyles }: TickTextProps) {
+  return (
+    <View style={[styles.tickText, { height: textHeight }]}>
+      <Text style={[styles.text, textStyles]}>{value}</Text>
+    </View>
+  )
+}
+
+function Tick({ startValue, endValue, textHeight, textStyles, animationDuration }: TickProps) {
+  const animatedValue = new Animated.Value(startValue * textHeight * -1)
+  const transformStyle = { transform: [{ translateY: animatedValue }] }
+  const duration = animationDuration ?? 1300
+
+  Animated.timing(animatedValue, {
+    toValue: endValue * textHeight * -1,
+    duration,
+    useNativeDriver: true,
+  }).start()
+
+  return (
+    <Animated.View style={[transformStyle]}>
+      {numberRange.map((number, index) => {
+        return (
+          <TickText key={index} textHeight={textHeight} textStyles={textStyles} value={number} />
+        )
+      })}
+    </Animated.View>
+  )
+}
+
+export default function NumberTicker({
+  finalValue,
+  textStyles,
+  textHeight,
+  animationDuration,
+  testID,
+}: Props) {
+  const finalValueArray = finalValue.toString().split('')
+
+  // For the startValueArray, map over each character in the finalValueArray to
```

Comment:
perhaps we could use [accessibility properties](https://reactnative.dev/docs/accessibility#accessible) `accessible` and `accessibilyLabel` to ensure that we have the actual number for the screen readers and unit tests?

Reply:
how does this work? would we have to apply custom logic for the values of accessible and accessibilyLabel based on the animation status? (i think we don't get a value like "isVisible" or "isAnimating" after rendering in a unit test environment). i thought about testing with accessiblity labels but ultimately decided that if i need to apply logic to set the values, then i'd be testing that logic rather than the animation. we're also not prioritising accessibility and screen readers (i.e. we're not designing with this in mind and we have not added any type of testing for these topics) so i don't really feel the need to go through extra effort to make sure this fun animation works with screen readers....

Reply:
I was thinking of just passing through the final value straight to the `accessibilityLabel`, so we can search for it in the tests, ensuring the element is rendered with this value.

Reply:
i think i prefer no test coverage to test coverage that doesn't directly test the outcome (false sense of security makes us more reckless). i was thinking that (import) animations should be tested manually instead, perhaps in our release testing steps